frontend/src/scenes/organization/Settings/inviteLogic.ts

```diff
@@ -0,0 +1,112 @@
+import { organizationLogic } from 'scenes/organizationLogic'
+import { eventUsageLogic } from 'lib/utils/eventUsageLogic'
+import { inviteLogicType } from './inviteLogicType'
+import { preflightLogic } from 'scenes/PreflightCheck/logic'
+import { successToast } from 'lib/utils'
+import { router } from 'kea-router'
+import { urls } from 'scenes/urls'
+
+/** State of a single invite row (with input data) in bulk invite creation. */
+interface InviteRowState {
+    target_email: string
+    first_name: string
+    isValid: boolean
+}
+
+const EMPTY_INVITE: InviteRowState = { target_email: '', first_name: '', isValid: true }
+
+export const inviteLogic = kea<inviteLogicType<InviteRowState>>({
+    path: ['scenes', 'organization', 'Settings', 'inviteLogic'],
+    actions: {
+        updateInviteAtIndex: (payload, index: number) => ({ payload, index }),
+        deleteInviteAtIndex: (index: number) => ({ index }),
+        appendInviteRow: true,
+        resetInviteRows: true,
+    },
+    connect: {
+        values: [preflightLogic, ['preflight']],
+    },
+    reducers: {
+        invitesToSend: [
+            [EMPTY_INVITE] as InviteRowState[],
+            {
+                updateInviteAtIndex: (state, { payload, index }) => {
+                    const newState = [...state]
+                    newState[index] = { ...state[index], ...payload }
+                    return newState
+                },
+                deleteInviteAtIndex: (state, { index }) => {
+                    const newState = [...state]
+                    newState.splice(index, 1)
+                    return newState
+                },
+                appendInviteRow: (state) => [...state, EMPTY_INVITE],
+                resetInviteRows: () => [EMPTY_INVITE],
+            },
+        ],
+    },
+    selectors: {
+        canSubmit: [
+            (selectors) => [selectors.invitesToSend],
+            (invites: InviteRowState[]) =>
+                invites.filter(({ target_email }) => !!target_email).length > 0 &&
+                invites.filter(({ isValid }) => !isValid).length == 0,
+        ],
+    },
+    loaders: ({ values }) => ({
+        _invitedTeamMembers: [
+            [] as OrganizationInviteType[],
+            {
+                inviteTeamMembers: async () => {
+                    if (!values.canSubmit) {
+                        return { invites: [] }
+                    }
+
+                    const payload: Pick<OrganizationInviteType, 'target_email' | 'first_name'>[] =
+                        values.invitesToSend.filter((invite) => invite.target_email)
+                    eventUsageLogic.actions.reportBulkInviteAttempted(
+                        payload.length,
+                        payload.filter((invite) => !!invite.first_name).length
+                    )
+
+                    return await api.create('api/organizations/@current/invites/bulk/', payload)
+                },
+            },
+        ],
+        invites: {
```

Comment:
Inside one file, it's best to keep to one style from `loaders: { a: [def, {...}] }` vs `loaders: { a: { __default: def, ... } }`. I personally prefer the first one that's similar to the style used in reducers.

Reply:
FYI leftover code from moving the file, but good call out, thanks!